packages/iov-core/src/jsonrpcsigningserver.ts

```diff
@@ -0,0 +1,184 @@
+import { ChainId, UnsignedTransaction } from "@iov/bcp-types";
+import {
+  isJsonCompatibleDictionary,
+  jsonRpcCodeInvalidParams,
+  jsonRpcCodeInvalidRequest,
+  jsonRpcCodeMethodNotFound,
+  jsonRpcCodeServerErrorDefault,
+  JsonRpcErrorResponse,
+  JsonRpcRequest,
+  JsonRpcResponse,
+  parseJsonRpcId,
+  parseJsonRpcRequest,
+} from "@iov/jsonrpc";
+
+import { SigningServerCore } from "./signingservercore";
+
+interface RpcCallGetIdentities {
+  readonly name: "getIdentities";
+  readonly reason: string;
+  readonly chainIds: ReadonlyArray<ChainId>;
+}
+
+interface RpcCallSignAndPost {
+  readonly name: "signAndPost";
+  readonly reason: string;
+  readonly transaction: UnsignedTransaction;
+}
+
+type RpcCall = RpcCallGetIdentities | RpcCallSignAndPost;
+
+class ParamsError extends Error {}
```

Comment:
Custom classes to allow checks by client, right?

Fine as is, but wondering what is best practice? I have seen sometimes errors with special members eg `err.notFound` to signify a particular error type. I guess `instanceof` on a class is more modern?

Reply:
It is used internally only. I need to convert different exceptions to different JSON-RPC error codes.

Both ways work. I choose the way you do it in all typed languages, that is unfortunatly poorly supported by JavaScript, since you cannot catch by type directly. Using properties on Error should work as long as you don't use TypeScript, which disallows adding new members. Using TypeScript you need a custom type anyway. So yes, we could also have a single custom type that holds the type information.

Since the scope of this is only a single file, we can change at any time if we have a reason to